crates/rspack_plugin_runtime/src/runtime_module/get_chunk_filename.rs

```diff
@@ -117,64 +117,57 @@ impl RuntimeModule for GetChunkFilenameRuntimeModule {
         }
       });
 
-    let mut dynamic_filename: Option<&Filename> = None;
+    let mut dynamic_filename: Option<&str> = None;
     let mut max_chunk_set_size = 0;
-    let mut chunk_filenames = IndexMap::new();
+    let mut chunk_filenames = IndexMap::<&Filename, IndexSet<&ChunkUkey>>::new();
     let mut chunk_map = IndexMap::new();
 
     if let Some(chunks) = chunks {
       chunks
         .iter()
         .filter_map(|chunk_ukey| get_chunk_from_ukey(chunk_ukey, &compilation.chunk_by_ukey))
         .for_each(|chunk| {
-          let filename_template = (self.filename_for_chunk)(chunk, compilation);
+          let filename = (self.filename_for_chunk)(chunk, compilation);
 
-          if let Some(filename_template) = filename_template {
+          if let Some(filename) = filename {
             chunk_map.insert(&chunk.ukey, chunk);
 
-            let chunk_set = chunk_filenames
-              .entry(filename_template.clone())
-              .or_insert(IndexSet::new());
+            let chunk_set = chunk_filenames.entry(filename).or_insert(IndexSet::new());
 
             chunk_set.insert(&chunk.ukey);
-
-            let should_update = match dynamic_filename {
-              Some(dynamic_filename) => match chunk_set.len().cmp(&max_chunk_set_size) {
-                Ordering::Less => false,
-                Ordering::Greater => true,
-                Ordering::Equal => match filename_template
-                  .template()
-                  .len()
-                  .cmp(&dynamic_filename.template().len())
-                {
+            if let Some(filename_template) = filename.template() {
```

Comment:
Related implementation in webpack: https://github.com/webpack/webpack/blob/e4fee1527b5444da931be6694227143f1cba53c4/lib/runtime/GetChunkFilenameRuntimeModule.js#L65